common/sai_dataplane/snappi/sai_snappi_dataplane.py

```diff
@@ -36,8 +36,12 @@ def init(self):
         # IxNetwork:    port.location = "<chassisip>;card;port"
         # TRex:         port.location = "localhost:5555"
         for pid, port in enumerate(self.config['port_groups']):
-            location = port.get('location', f"localhost:{BASE_TENGINE_PORT+pid}")
```

Comment:
Since `location` should be defined in both cases - ixnet and ixia-c - I'd suggest just to add an assertion something like this:
```
location = port.get('location', f"localhost:{BASE_TENGINE_PORT+pid}")
if self.mode == 'ixnet':
    assert not location.startswith("localhost"), f"Invalid {self.mode} port location {location}"
self.configuration.ports.port(name=port["name"], location=location)
```
This will help to get meaningful error message in case of misconfiguration.

Reply:
this has been fixed